soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -7095,16 +7105,39 @@ s32 Player_TryLeavingCrawlspace(Player* this, PlayState* play) {
 
             if (this->linearVelocity > 0.0f) {
                 this->actor.shape.rot.y = this->actor.wallYaw + 0x8000;
-                Player_AnimPlayOnce(play, this, &gPlayerAnim_link_child_tunnel_end);
-                Player_AnimReplaceApplyFlags(play, this, 0x9D);
-                OnePointCutscene_Init(play, 9601, 999, NULL, MAIN_CAM);
+                // #region SOH [Enhancement]
+                if (CVarGetInteger("gCrawlSpeed", 1) > 1) {
+                    // animation when exiting a tunnel forward
+                    LinkAnimation_Change(play, &this->skelAnime, &gPlayerAnim_link_child_tunnel_end,
+                                         1.0f * ((CVarGetInteger("gCrawlSpeed", 1) + 1.0f) / 2.0f), 0.0f,
```

Comment:
Why the `+ 1.0f) / 2.0f`?

Would be simpler with
```suggestion
                                         (CVarGetInteger("gCrawlSpeed", 1)), 0.0f,
```

Reply:
My question for this would be why store something in an integer if it's going to be a float? We have CVarGetFloat, and we have clamp capability for ImGui sliders.

Reply:
A bunch of other properties are also used as integers despite being used on floating variables like for example `gClimbSpeed` or `gFasterBlockPush`. I think consistency prioritize here but if `gCrawlSpeed` needs to be a float, the change is required for the other properties at the same time

Reply:
1) In testing, scaling the animations at the same rate as the crawl speed led to extreme animation speeds and then a more expected increase in crawl speed. This scaling on the animations looked more reasonable to me. I'm not tied to this.
2) I couldn't find a way to increment the float sliders by a full integer value via the +/- buttons without modifying them or adding a new one. Using the int slider (which also had precedent) seemed less intrusive.
3) The integer value is used for the other piece, though not a dealbreaker.

Reply:
So you're saying it's used in both int and float situations in code?

Reply:
Yes - in z_onepointdemo, it's used as an int to modify the cutscene to avoid camera swings coming out of the crawl animation. Both here, for the animation change, and below, for the actual crawl speed itself, it's being converted to a float.

Reply:
Ah, OK. Well, if the CVar value stored is never a non-whole-number, then integer would still make sense. But, to bring it back to Pierre's original question, would the conversion to float not be handled by the `/ 2.0f`, making the initial `1.0f *` redundant?

Reply:
Yes - that is a remnant of starting with the exiting tunnel backward section (it starts with -1.0f to play in reverse) and keeping the format, and can be removed.